src/engine/EventManager.cxx

```diff
@@ -409,6 +428,21 @@ void EventManager::checkEvents(SDL_Event &event, Engine &engine)
             {
               engine.map->highlightNode(highlitNode, SpriteHighlightColor::RED);
             }
+            const Point &buildingCoordinates =
+                engine.map->findNodeInMap(convertIsoToScreenCoordinates(highlitNode), Layer::BUILDINGS);
+
+            if (!contains(m_transparentBuildings, buildingCoordinates))
+            {
+              if (buildingCoordinates != UNDEFINED_POINT)
+              {
+                const TileData *tileData = engine.map->getMapNode(buildingCoordinates)->getTileData(Layer::BUILDINGS);
+                if (tileData && tileData->category != "Flora")
+                {
+                  (engine.map->getMapNode(buildingCoordinates))->setNodeTransparency((float)40 / 100);
+                  m_transparentBuildings.push_back(buildingCoordinates);
+                }
+              }
```

Comment:
I don't like this implementation as we are searching for the MapNode twice. Once to check if it contains it, and another time to find it.